File: backend/youtube/responses/videoResponse.py
from typing import Optional, List

from backend.core.responses.baseVideoResponse import BaseVideoResponse
from backend.youtube.responses.channelResponse import YoutubeChannelResponse


class YoutubeVideoResponse(BaseVideoResponse):
    youtubeId: str
    viewCount: int
    likeCount: int
    commentCount: int
    channel: Optional[YoutubeChannelResponse] = None
    description: Optional[str] = None
    youtubeUrl: Optional[str] = None
    tags: List[str] = []
    publishedAt: Optional[str] = None
    video_path: Optional[str] = None
    audio_path: Optional[str] = None

File: frontend/dto/videoResponse.ts
import { z } from "zod";
import { ChannelResponseSchema } from "./channelResponse";

export const VideoResponseSchema = z.object({
    provider: z.string(),
    publicId: z.string(),
    youtubeId: z.string(),
    name: z.string(),
    duration: z.number(),
    viewCount: z.number(),
    likeCount: z.number(),
    commentCount: z.number(),
    internalImageUrl: z.string().nullable(),
    channel: z.lazy(() => ChannelResponseSchema).nullable(),
    description: z.string().nullable(),
    youtubeUrl: z.string().nullable(),
    tags: z.array(z.string()),
    publishedAt: z.string().nullable(),
});

export type VideoResponse = z.infer<typeof VideoResponseSchema>;
